src/PlatesEngineFactory.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @see       https://github.com/zendframework/zend-expressive-platesrenderer for the canonical source repository
+ * @copyright Copyright (c) 2016 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-platesrenderer/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\Expressive\Plates;
+
+use Interop\Container\ContainerInterface;
+use League\Plates\Engine as PlatesEngine;
+use League\Plates\Extension\ExtensionInterface;
+
+/**
+ * Create and return a Plates engine instance.
+ *
+ * Optionally uses the service 'config', which should return an array. This
+ * factory consumes the following structure:
+ *
+ * <code>
+ * 'plates' => [
+ *     'extensions' => [
+ *         // extension instances, or
+ *         // service names that return extension instances, or
+ *         // class names of directly instantiable extensions.
+ *     ]
+ * ]
+ * </code>
+ *
+ * By default, this factory attaches the Extension\UrlExtension to
+ * the engine. You can override the functions that extension exposes
+ * by providing an extension class in your extensions array, or providing
+ * an alternative Zend\Expressive\Plates\Extension\UrlExtension service.
+ */
+class PlatesEngineFactory
+{
+    /**
+     * @param ContainerInterface $container
+     * @return PlatesEngine
+     */
+    public function __invoke(ContainerInterface $container)
+    {
+        $config = $container->has('config') ? $container->get('config') : [];
+        $config = isset($config['plates']) ? $config['plates'] : [];
+
+        // Create the engine instance:
+        $engine = new PlatesEngine();
+
+        $this->injectUrlExtension($container, $engine);
```

Comment:
In here, instead of hardcoded inject the UrlExtesion, we could allow more extensions to be provided via config, like: 

```php
'templates' => [
        'extension' => 'phtml',
        'paths' => [
            'app' => ['templates/app'],
            'layout' => ['templates/layout'],
            'error' => ['templates/error'],
        ],
        'extensions' => [
             Extension\UrlExtension::class => Zend\Expressive\Plates\Extension\UrlExtensionFactory::class
        ],
    ]
```